js/popup-window.js

```diff
@@ -0,0 +1,215 @@
+
+cardWorks2.className = 'works-3';
+const workCard2 = document.createElement('article');
+const workCardDiv2 = document.createElement('div');
+workCardDiv2.className = 'card-works';
+cardWorks2.appendChild(workCard2);
+workCard2.className = 'card-works';
+workCard2.innerHTML = `<img src=${cardetails[2].img} alt="snapshootimg" class="snapshoot-img">
+<img src=${cardetails[2].imagedsk} alt="snapshootdsk" class="snapshoot-img-dsk">
+<div class="adjust">
+  <div class="leftblock">
+    <h2 class="card-works-title-1 popup-title">${cardetails[2].titledsk}</h2>
+    <h2 class="card-works-title-hidden">${cardetails[2].titledsk}</h2>
+    <div class="ref">
+      <span class="client-1">${cardetails[2].spans[0]}</span>
+      <span class="client-hidden">${cardetails[2].spansdsk[0]}</span>
+      <span><i class="fa fa-circle fa-xs"></i></span>
+      <span class="role">${cardetails[2].spans[1]}</span>
+      <span><i class="fa fa-circle"></i></span>
+      <span class="year">${cardetails[2].spans[2]}</span>
+    </div>
+    <p class="card-primary-text">${cardetails[2].textcontent}</p>
+  </div>
+  <ul class="tags">
+    <li class="tag">${cardetails[2].technologies[0]}</li>
+    <li class="tag">${cardetails[2].technologies[1]}</li>
+    <li class="tag-hidden">${cardetails[2].technologiesdsk[2]}</li>
+    <li class="tag">${cardetails[2].technologies[2]}</li>
+  </ul>
+  <div class="ref">
+    <button class="popup-btn" id="3">${cardetails[2].btntxt}</button>
+  </div>
+</div>`;
+
+cardWorks3.className = 'works-4';
+const workCard3 = document.createElement('article');
+const workCardDiv3 = document.createElement('div');
+workCardDiv3.className = 'card-works';
+cardWorks3.appendChild(workCard3);
+workCard3.className = 'card-works';
+workCard3.innerHTML = `<img src=${cardetails[3].img} alt="snapshootimg" class="snapshoot-img">
+<div class="adjust">
+  <div class="leftblock">
+    <h2 class="card-works-title-1 popup-title">${cardetails[3].titledsk}</h2>
+    <h2 class="card-works-title-hidden-1">${cardetails[3].titledsk}</h2>
+    <div class="ref">
+      <span class="client-1">${cardetails[3].spans[0]}</span>
+      <span class="client-hidden-2">${cardetails[3].spansdsk[0]}</span>
+      <span><i class="fa fa-circle fa-xs"></i></span>
+      <span class="role">${cardetails[3].spans[1]}</span>
+      <span><i class="fa fa-circle"></i></span>
+      <span class="year">${cardetails[3].spans[1]}</span>
+    </div>
+    <p class="card-primary-text">${cardetails[3].textcontent}</p>
+  </div>
+  <ul class="tags">
+    <li class="tag">${cardetails[3].technologies[0]}</li>
+    <li class="tag">${cardetails[3].technologies[1]}</li>
+    <li class="tag-hidden">${cardetails[3].technologiesdsk[2]}</li>
+    <li class="tag">${cardetails[3].technologies[2]}</li>
+  </ul>
+  <div class="ref">
+    <button class="popup-btn" id="4">${cardetails[2].btntxt}</button>
+  </div>
+</div>
+<img src=${cardetails[3].imagedsk} alt="snapshoot-img" class="snapshoot-img-dsk-2">`;
```

Comment:
- [ ] Although your work was impressive, there is a requirement to use JavaScript to dynamically generate work cards :+1:. Although you have already created the necessary elements with JavaScript, you must now utilize a loop to dynamically generate the cards, populating them with the information stored in the array of objects. This is necessary to ensure that the work cards are generated dynamically and accurately reflect the information in the array.